internal/pkg/service/buffer/worker/service/cleanup.go

```diff
@@ -0,0 +1,90 @@
+package service
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/worker/distribution"
+)
+
+type cleanup struct {
+	*Service
+	logger   log.Logger
+	assigner *distribution.Assigner
+
+	lock *sync.RWMutex
+}
+
+func (s *Service) cleanup(ctx context.Context, wg *sync.WaitGroup) <-chan error {
+	return s.dist.StartWork(ctx, wg, s.logger, func(ctx context.Context, assigner *distribution.Assigner) (initDone <-chan error) {
+		return startCleanup(ctx, wg, s, assigner)
+	})
+}
+
+func startCleanup(ctx context.Context, wg *sync.WaitGroup, s *Service, assigner *distribution.Assigner) <-chan error {
+	c := &cleanup{
+		Service:  s,
+		logger:   s.logger.AddPrefix("[cleanup]"),
+		assigner: assigner,
+		lock:     &sync.RWMutex{},
+	}
+
+	// Start ticker
+	startTime := c.clock.Now()
+	c.startTicker(ctx, wg)
+	c.logger.Infof(`initialized | %s`, c.clock.Since(startTime))
```

Comment:
`startTicker` iba spusta goroutine, takze toto meranie casu tu nema zmysel.

V mieste odkial si to zobral sa toho pravdepodobne dialo viac, alebo je chyba aj tam.

To `initialized | time` som zvykol logovat, ked sa nieco taha na zaciatku z etcd cez `GetAllAndWatch`, ak by tam bol do buducna nejaky problem s rychlostou, tak ze to hned vidime.